### ISSUE TITLE
Why is nasnet model so slow using one gpu?

### ISSUE THREAD

**Issue by chenghuige** [tensorflow/models#2763]
tensorflow 1.4 rc0,  on p40
I'm tring nasnet, it performs excellent but one thing is curious, I find using one gpu is much slower then using 2gpu(tower loss) I'm using p40, say if one gpu batch size is 16, for 2 gpu, each gpu with batch size 16 (so batch size total 16 * 2) But since we have communication cost I suppose using 2gpu will not speed up * 2, However experiment result is like below

1 gpu: train_step:92700 duration:37.092 elapsed:[405.920] batch_size:[16] batches/s:[0.25] insts/s:[3.94] 1epoch:[74.00h]

2 gpu: train_step:174600 duration:47.933 elapsed:[236.972] batch_size:[16] gpus:[2] batches/s:[0.42] insts/s:[13.50] 1epoch:[21.60h]

I have also try using inceptionV4, resnet152, inceptionResnetV2, they all behave as expected, 2gpu not speed up *2.

Sometimes I prefer to use one gpu, since when finetune image model, multiple gpu tower loss might hurt final accuracy though I still know why..

Not sure if this is a bug, I have also asked quesion on stackoverlow. 
https://stackoverflow.com/questions/47215223/why-is-nasnet-model-so-slow-using-one-gpu
Please help close if it is not proper to discuss here.

**Comment by shlens** [comment]
I am sorry you are having such trouble with NASNet. I have a few clarification questions that are not clear from your question:

What is the average (over multiple iterations) duration it takes to run 1 training batch for 1 GPU with batch size 16?

What is the average (over multiple iterations) duration it takes to run 1 training batch for 2 GPUs where each GPU contains batch size 8? (This means that the total batch size of the training run is 16).

**Comment by chenghuige** [comment]
@shlens  Well I noticed there might be some perf issue due to running on p40 cluster reading from hdfs.
So I changed to test on local gtx1080ti machine, due to 11G mem, I can only run batch size 8 for each gpu at most. I just tested with using 1 gpu with batch size 8 and using 2 gpu with each gpu batch size 4.

Using 1 gpu, batch size 8
one batch took **2.08 s**,  batch_size:[8] batches/s:[0.48] insts/s:[3.87]  1epoch:[3.86h]
Using 2 gpu, batch size 4 * 2
one batch took **0.617s** batch_size:[4] gpus:[2] batches/s:[1.62] insts/s:[12.94]  1epoch:[1.16h]

Still using 2 gpu is much faster, from 1 epoch using 3.86 hour to only 1.16 hour

**Comment by shlens** [comment]
@chenghuige I am surprised by this behavior. Would you be able to run with batch size = 8, 16, 32 on 1 GPU to see if the scaling with batch is as expected?

**Comment by chenghuige** [comment]
@shlens  Well due to mem limit I can not run batch size 16 on gtx1080ti card. 
I tested batch size 4 as below, one batch about 0.5 s
batch_size:[4] batches/s:[1.99] insts/s:[7.98]  1epoch:[1.88h]

**Comment by cy89** [comment]
@shlens any other observations here? Is there any advice to offer @chenghuige?

**Comment by shlens** [comment]
We will not have time investigate this further but I would suggest that @chenghuige examine a timeline profile of the function to potentially identify bottleneck in speed.

**Comment by chenghuige** [comment]
Ok @shlens  thanks, currently I find 2gpu can still have good finetune result so just use 2gpu running. I will examine more when I have time.

**Comment by muayyad-alsadi** [comment]
Would you please check if this is the reason casing nasnet slower than expected

https://stackoverflow.com/questions/48454379/why-does-last-layer-in-nasnet-use-fully-connected-instead-of-1x1-conv2d

**Comment by clintgt** [comment]
Anyone had success with speeding up NASNet?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
